File: php/pig-latin/pig-latin.php
<?php

function pig_word($matches){
    $word = $matches[0];
    $alpha = implode("",range("a","z"));
    $voyelle = "aeiouy";
    $consonne = preg_replace("/[$voyelle]/","",$alpha);

    preg_match(
        "/^(squ|qu|y|[$consonne]+)([$voyelle].*)$/",
        $word,
        $matches
    );

    if(
        $matches
        && !preg_match("/ay$/",$word)
    ){
        return $matches[2].$matches[1]."ay";
    }else{
        return $word . "ay";
    }
}

function translate(string $words){
    return preg_replace_callback("/(\w+)/","pig_word", $words);
}

File: python/pig-latin/pig_latin.py
import re

VOYELLE = "aeiouy"
CONSONNE = "bcdfghjklmnpqrstvwxz"


def translate(text):
    # rules 2+3+4
    pattern = "^(squ|qu|y|[" + CONSONNE + "]+)([" + VOYELLE + "].*)$"
    words = []
    for word in text.split(" "):
        mc = re.search(pattern, word)
        if mc and not re.search(r"ay$", word):
            pig_word = mc.group(2) + mc.group(1) + "ay"
        else:
            # rules 1
            pig_word = word + "ay"
        words.append(pig_word)
    return " ".join(words)
